src/libstd/io/stdio.rs

```diff
@@ -175,13 +175,17 @@ impl Read for Stdin {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<'a> Read for StdinLock<'a> {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        self.inner.read(buf)
+        self.inner.borrow_mut().read(buf)
     }
 }
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<'a> BufRead for StdinLock<'a> {
-    fn fill_buf(&mut self) -> io::Result<&[u8]> { self.inner.fill_buf() }
-    fn consume(&mut self, n: usize) { self.inner.consume(n) }
+    fn fill_buf(&mut self) -> io::Result<&[u8]> {
+        self.inner.borrow_mut().fill_buf()
```

Comment:
I have an odd lifetime error, I don’t understand and can’t find a way to fix. Help appreciated.